tokio-console/src/config.rs

```diff
@@ -110,17 +113,57 @@ pub struct ViewOptions {
 #[derive(Clap, Debug, Copy, Clone)]
 pub struct ColorToggles {
     /// Disable color-coding for duration units.
-    #[clap(long = "no-duration-colors", parse(from_flag = std::ops::Not::not), group = "colors")]
-    pub(crate) color_durations: bool,
+    #[clap(long = "no-duration-colors", group = "colors")]
+    color_durations: Option<bool>,
 
     /// Disable color-coding for terminated tasks.
-    #[clap(long = "no-terminated-colors", parse(from_flag = std::ops::Not::not), group = "colors")]
-    pub(crate) color_terminated: bool,
+    #[clap(long = "no-terminated-colors", group = "colors")]
+    color_terminated: Option<bool>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct ConfigFile {
+    charset: Option<CharsetConfig>,
+    colors: Option<ColorsConfig>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct CharsetConfig {
+    lang: String,
+    ascii_only: bool,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct ColorsConfig {
+    enabled: bool,
+    truecolor: bool,
+    palette: Palette,
+    enable: ColorsEnable,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct ColorsEnable {
+    durations: bool,
+    terminated: bool,
 }
```

Comment:
Ok, I'll try it. thank you!

Reply:
> this struct seems very similar to ColorToggles, is it really necessary to have separate ColorToggles and ColorsEnable structs?

No, it does not necessary. I think it can be fixed as you suggested.
I made a mistake that `durations` and `terminated` are required in the config file.

> can we just derive Deserialize for ColorToggles, and continue inverting the command-line arguments as part of the parse function when parsing CLI args?

I think to invert the bool value with the getter method. 

```
// === impl ColorToggles ===

impl ColorToggles {
    pub fn color_durations(&self) -> bool {
        self.color_durations.map(std::ops::Not::not).unwrap_or(true)
    }

    pub fn color_terminated(&self) -> bool {
        self.color_durations.map(std::ops::Not::not).unwrap_or(true)
    }
}
```
---

**Why I remove `parse(from_flag = std::ops::Not::not)` attribute.**

First, I tried this approach.

I changed the fields (color_durations, color_terminated) to optional.
But, this approach does not pass the compile. because `from_flag` required `fn(bool) -> T`.

```rust
#[derive(Clap, Debug, Copy, Clone)]
pub struct ColorToggles {
    /// Disable color-coding for duration units.
    #[clap(long = "no-duration-colors", parse(from_flag = std::ops::Not::not), group = "colors")]
    pub(crate) color_durations: Option<bool>,

    /// Disable color-coding for terminated tasks.
    #[clap(long = "no-terminated-colors", parse(from_flag = std::ops::Not::not), group = "colors")]
    pub(crate) color_terminated: Option<bool>,
}
```

So I defined the getter method. 

```
// === impl ColorToggles ===

impl ColorToggles {
    pub fn color_durations(&self) -> bool {
        self.color_durations.map(std::ops::Not::not).unwrap_or(true)
    }

    pub fn color_terminated(&self) -> bool {
        self.color_durations.map(std::ops::Not::not).unwrap_or(true)
    }
}
```

Reply:
Okay, that seems fine to me, then!